### PR TITLE
AF-1837 dart 1 and dart2 compat

### PR DESCRIPTION
**Pull request by robbecker-wf**
# Overview
Prepare the code and process to work under both dart 1 and dart 2

# Testing
 - [ ] CI should pass under both dart 1 and dart 2

### CODE REVIEW

**Comment by aviary2-wf** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://dev.workiva.net/docs/information-security/aviary#configuring-aviary) coverage of security relevant code
---
_Questions or Comments? Reach out on HipChat: InfoSec Forum._

File: Dockerfile [hunk 1]
```diff
@@ -15,20 +15,19 @@ WORKDIR /build/
...
-		xvfb-run -s '-screen 0 1024x768x24' pub run dart_dev coverage --no-html && \
-		pub run dart_dev docs --no-open && \
-		tar czvf api.tar.gz -C doc/api . && \
-		pub build test && \
-		./tool/stage_for_cdn.sh && \
-		tar -hcf build.tar.gz build/test/ && \
-		tar czvf font_face_observer.pub.tgz LICENSE README.md pubspec.yaml lib/ && \
-		./tool/codecov.sh && \
-		echo "Script sections completed"
+	pub get && \
```
Comment (robbecker-wf) [inline Dockerfile]: remove --packages-dir

File: Dockerfile [hunk 2]
```diff
@@ -15,20 +15,19 @@ WORKDIR /build/
...
+	pub publish --dry-run && \
+	pub run dart_dev analyze && \
+	pub run dart_dev format --check && \
+	xvfb-run -s '-screen 0 1024x768x24' pub run dart_dev test --web-compiler=dartdevc -p chrome && \
```
Comment (robbecker-wf) [inline Dockerfile]: removed coverage since there isn't a way to collect coverage under dart 2 yet

File: lib/data_uri.dart [hunk 3]
```diff
@@ -61,9 +61,10 @@ Copyright © [YEAR] W3C® (MIT, ERCIM, Keio, Beihang)."
 
 https://www.w3.org/Consortium/Legal/2015/copyright-software-and-document
 */
-import 'dart:convert';
+
 import 'dart:html';
 import 'dart:typed_data';
+import 'package:dart2_constant/convert.dart' as convert;
```
Comment (robbecker-wf) [inline lib/data_uri.dart]: use dart2_constant to use dart 1&2 compatible constant names

File: lib/src/adobe_blank.dart [hunk 4]
```diff
@@ -66,7 +66,7 @@ https://www.w3.org/Consortium/Legal/2015/copyright-software-and-document
 const String adobeBlankFamily = 'AdobeBlank';
 
 /// The font key for the AdobeBlank font
-final String adobeBlankKey = '${adobeBlankFamily}_normal_normal_normal';
+const String adobeBlankKey = '${adobeBlankFamily}_normal_normal_normal';
```
Comment (robbecker-wf) [inline lib/src/adobe_blank.dart]: lint

File: test/font_face_observer_test.dart [hunk 5]
```diff
@@ -216,7 +216,7 @@ void main() {
 
 
     test('should keep data-uses attribute up to date', () async {
-      final String differentGroup = 'diff';
+      const String differentGroup = 'diff';
```
Comment (robbecker-wf) [inline test/font_face_observer_test.dart]: lint ^

File: test/index.html [hunk 6]
```diff
@@ -1,18 +1,23 @@
...
-        padding: 5px;
-      }
+    #table,
+    #table tr,
+    #table th,
+    #table td {
+      border: solid 1px #ccc;
+      border-collapse: collapse;
+      padding: 5px;
+    }
```
Comment (robbecker-wf) [inline test/index.html]: just format changes

File: test/index.html [hunk 7]
```diff
@@ -25,7 +30,7 @@
       <th>DOM</th>
       <th>Canvas</th>
     </tr>
-  <script type="application/dart" src="main.dart"></script>
-  <script src="../packages/browser/dart.js"></script>
+    <script defer src="main.dart.js"></script>
```
Comment (robbecker-wf) [inline test/index.html]: get rid of transformers, browser package according to https://webdev-dartlang-org-dev.firebaseapp.com/dart-2

File: test/main.dart [hunk 8]
```diff
@@ -164,24 +164,24 @@ Future<Null> _loadFont(_FontConfig cfg, bool useSimulatedLoadEvents) async {
...
+    ..append(new TableCellElement()..text = result.isLoaded ? 'Yes' : 'No')
+    ..append(new TableCellElement()
+      ..text = pass ? 'Yes' : 'No'
+      ..style.backgroundColor = pass ? 'green' : 'red')
+    ..append(new TableCellElement()
+      ..append(new DivElement()
+        ..text = message
+        ..style.fontFamily = uniqFamily
+        ..style.fontSize = '18px'))
+    ..append(new TableCellElement()..append(canvas));
```
Comment (robbecker-wf) [inline test/main.dart]: format

File: test/unload.html [hunk 9]
```diff
@@ -30,7 +30,6 @@ <h1>Memory / Perf Analysis</h1>
   <div style="font-family: Roboto_2, Times New Roman">Roboto_2 group_A &amp; group_B</div>
   <div style="font-family: Roboto_3, Times New Roman">Roboto_3 group_A</div>
   
-  <script type="application/dart" src="unload.dart"></script>
-  <script src="../packages/browser/dart.js"></script>
+  <script defer src="unload.dart.js"></script>
```
Comment (robbecker-wf) [inline test/unload.html]: Changes for Dart 2 / DDC .. no rewriting, no transformers

File: Dockerfile [hunk 10]
```diff
@@ -15,20 +15,29 @@ WORKDIR /build/
...
-		tar -hcf build.tar.gz build/test/ && \
-		tar czvf font_face_observer.pub.tgz LICENSE README.md pubspec.yaml lib/ && \
-		./tool/codecov.sh && \
-		echo "Script sections completed"
+	wget https://storage.googleapis.com/dart-archive/channels/dev/release/2.0.0-dev.58.0/sdk/dartsdk-linux-x64-release.zip && \
+	unzip dartsdk-linux-x64-release.zip && \
+	export D2PATH=`pwd`/dart-sdk/bin && \
+	export PATH=$D2PATH:$PATH && \
+	dart --version && \
+	pub --version && \
```
Comment (robbecker-wf) [inline Dockerfile]: Download dart 2 dev, extract and add it to the path to use dart 2 in CI. Maybe we could include both versions of Dart this way in CI images include a quick command to switch versions easily?

File: lib/data_uri.dart (hunk 3 above)
Comment (joelleibow-wf) [reply lib/data_uri.dart]: For my own edification, is there a reference you can provide explaining this change?

File: test/font_face_observer_test.dart [hunk 11]
```diff
@@ -103,12 +103,12 @@ void main() {
     });
 
     test('should init correctly with passed in values', () {
-      final String family = 'my family';
-      final String style = 'my style';
-      final String weight = 'my weight';
-      final String stretch = 'my stretch';
-      final String testString = 'my testString';
-      final int timeout = 1337;
+      const String family = 'my family';
```
Comment (joelleibow-wf) [inline test/font_face_observer_test.dart]: Is `const` the new Dart 2 preferred?

File: lib/data_uri.dart (hunk 3 above)
Comment (robbecker-wf) [reply lib/data_uri.dart]: There's a library that allows you to use constants that work under both dart 1 and dart 2 https://pub.dartlang.org/packages/dart2_constant
If we didn't care about backwards compatibility, we could just move to dart 2 camelcase constants by running https://pub.dartlang.org/packages/dart2_fix

File: test/font_face_observer_test.dart (hunk 11 above)
Comment (robbecker-wf) [reply test/font_face_observer_test.dart]: Yes, Dart 2 now recommends you use const in certain places. It isn't an error, just a lint.

**Comment by corwinsheahan-wf** [comment]
QA +1 / +10

- [x] Testing instruction
- [x] Dev +1's
- [x] Dev/QA +10 with detail of what was tested:
    * CI Passes
- [x] Security review (if required)
- [x] Unit tests created/updated
- [x] All unit tests pass
- [x] Rosie has run and reports properly the release the ticket will be included in

Merging into master.
@Workiva/release-management-p

**Pull request merged**